pkg/action/install.go

```diff
@@ -456,6 +457,20 @@ func (c *Configuration) renderResources(ch *chart.Chart, values chartutil.Values
 		}
 	}
 
+	if outputHooks {
```

Comment:
While I totally see why this is here (install and template use the same code), this modification to the function feels like it doesn’t belong here and should be a `helm template` only implementation. I think someone could accidentally inject the hooks into a manifest they are installing (if using Helm as a library). So I think this code should live in `cmd/helm/template.go` and then we can export the WriteToFile as a helper. Any thoughts on this idea?

Reply:
Hey! Thanks for your feedback.

> So I think this code should live in cmd/helm/template.go

Yeah that makes sense! However, I'm not entirely sure how it can be implemented without refactoring/duplicating existing code.

What we need to achieve is that do it without modifying `func (i *Install) Run`, `renderResources`, and `runInstall`, right?

To me that basically mean we need to implement something like `runTemplate` which calls the existing `renderResources` and this block of code specific to `template`(for rendering hooks). Most of `runTemplate` code will be duplicated from `runInstall`.

Is that okay? Or would you like to me to factor common parts of `runInstall` and `runTemplate`?

Reply:
You are totally right. What if we just did this in `cmd/helm/template.go` with 

```go
// Preexisting line for reference
fmt.Fprintln(out, strings.TrimSpace(rel.Manifest))
if !client.DisableHooks {
	for _, h := range rel.Hooks {
		fmt.Fprintln(out, h.Manifest)
	}
}
```

It would avoid adding an extra flag and no need to refactor the underlying client

Reply:
That sounds great! I now realize that the release object returned from `runInstall` has what I needed. I'll try and report back soon.